stock_reception_screen/models/stock_reception_screen.py

```diff
@@ -124,8 +125,21 @@ class StockReceptionScreen(models.Model):
     current_move_product_vendor_code = fields.Char(
         related="current_move_id.vendor_code"
     )
+    current_move_product_qty_available = fields.Float(
+        related="current_move_id.product_id.qty_available"
+    )
+    current_move_product_outgoing_qty = fields.Float(
+        related="current_move_id.product_id.outgoing_qty"
+    )
+    current_move_product_last_lot_life_date = fields.Datetime(
+        compute="_compute_current_move_product_last_lot_life_date",
+        string="Most recent exp. date",
+    )
     # current move line
     current_move_line_id = fields.Many2one(comodel_name="stock.move.line", copy=False)
+    current_move_line_original_location_dest_id = fields.Many2one(
+        related="current_move_line_id.original_location_dest_id"
+    )
```

Comment:
```suggestion
    current_move_line_original_location_mapping = Serialized(default={})
    current_move_line_original_location_dest_id = fields.Many2one(
        compute="_compute_current_move_line_original_location_dest_id"
    )
```
In the compute lookup for the mapping that you build on create instead of storing the value on the move line itself.
What do you think?

Reply:
I prefered to store the initial destination location to always propose it to the user (see the domain on the view) even if there are other places where to store the goods, because I suppose the user still don't know if the other places are free, so better to still propose the default value.
And it was store on create because when processing a line on the screen, we could change the destination, then cancel the screen. At this stage we have lost the original value computed by Odoo.

But by writing it maybe you're right: like the package, perhaps we should wait the validation of all the steps to make these data persistant on the move line... it'll require more work to do so.

Reply:
I changed the implementation and dropped this field.